staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -45,6 +50,194 @@ func TestDecode(t *testing.T) {
 		want          interface{}
 		assertOnError func(t *testing.T, e error)
 	}{
+		{
+			name: "reject duplicate keys (int)",
+			in:   hex("a201010102"), // {1:1, 1:2}
+			into: map[int64]interface{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if e.Index != 1 {
+					t.Errorf("expected %#v, got %#v", &cbor.DupMapKeyError{Index: 1}, e)
+				}
+			}),
+		},
+		{
+			name: "reject duplicate keys (text tring) (map)",
```

Comment:
Nit: Easier to filter with `-run` if subtest names don't contain parens.

```suggestion
			name: "reject duplicate text string keys into map",
```